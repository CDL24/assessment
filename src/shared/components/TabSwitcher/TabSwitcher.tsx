import React, { useMemo, useState } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { useTheme } from "@react-navigation/native";
import createStyles from "./TabSwitcherStyle";

type Props = {
  tab1: string;
  tab2: string;
}
const TabSwitcher: React.FC<Props> = ({tab1, tab2}) => {
  const theme = useTheme();
  const styles = useMemo(() => createStyles(theme), [theme]);

  const [activeTab, setActiveTab] = useState<number>(1);

  const handleTabPress = (tabIndex: number) => {
    setActiveTab(tabIndex);
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={[styles.tab, activeTab === 1 ? styles.activeTab : null]}
        onPress={() => handleTabPress(1)}
        testID='Tab1'
      >
        <Text style={activeTab === 1 ? styles.tabTextActive : styles.tabText}>{tab1}</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.tab, activeTab === 2 ? styles.activeTab : null]}
        onPress={() => handleTabPress(2)}
        testID='Tab2'
      >
        <Text style={activeTab === 2 ? styles.tabTextActive : styles.tabText}>{tab2}</Text>
      </TouchableOpacity>
    </View>
  );
};
export default TabSwitcher;
