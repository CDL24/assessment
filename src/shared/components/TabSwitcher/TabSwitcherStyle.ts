import { StyleSheet, TextStyle, ViewStyle } from "react-native";
import { ExtendedTheme } from "@react-navigation/native";
import { moderateScale, verticalScale } from "@theme/metrix";
import fonts from "@fonts";
import fontSize from "@font-size";

interface Style {
  container: ViewStyle;
  tab: ViewStyle;
  activeTab: ViewStyle;
  tabText: TextStyle;
  tabTextActive: TextStyle;
}

export default (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create<Style>({
    container: {
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: verticalScale(20),
    },
    tab: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingVertical: verticalScale(8),
      borderRadius: moderateScale(10)
    },
    activeTab: {
      backgroundColor: colors.primary,
      borderRadius: moderateScale(10),
    },
    tabText: {
      fontSize: fontSize.font12,
      fontFamily: fonts.poppins.bold,
      color: colors.primary,
    },
    tabTextActive: {
      fontSize: fontSize.font12,
      fontFamily: fonts.poppins.bold,
      color: colors.white,
    }
  });
};
