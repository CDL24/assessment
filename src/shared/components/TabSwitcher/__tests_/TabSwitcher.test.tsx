import React from 'react';
import {it} from '@jest/globals';
import { render } from '@testing-library/react-native';
import TabSwitcher from '../TabSwitcher';
import { create } from 'react-test-renderer';

describe("<TabSwitcher /> Component", ()=>{
    it("render correctly", ()=>{
        const tree = create(<TabSwitcher tab1='Tab1' tab2='Tab2' />).toJSON();
        expect(tree).toMatchSnapshot();       
    });

    it('onTab pressed', ()=>{
        const {getByText} = render(<TabSwitcher tab1='Tab1' tab2='Tab2' />);
        expect(getByText('Tab1')).toBeTruthy()
    })
});