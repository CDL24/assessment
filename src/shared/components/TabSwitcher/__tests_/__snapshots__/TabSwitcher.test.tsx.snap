// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabSwitcher /> Component render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 32.857142857142854,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 122, 255)",
        "borderRadius": 15,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 13.142857142857142,
      }
    }
    testID="Tab1"
  >
    <Text
      style={
        {
          "color": undefined,
          "fontFamily": "Poppins-Bold",
          "fontSize": 12,
        }
      }
    >
      Tab1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 15,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 13.142857142857142,
      }
    }
    testID="Tab2"
  >
    <Text
      style={
        {
          "color": "rgb(0, 122, 255)",
          "fontFamily": "Poppins-Bold",
          "fontSize": 12,
        }
      }
    >
      Tab2
    </Text>
  </View>
</View>
`;
